R/archive.R

```diff
@@ -145,14 +150,64 @@ epi_archive =
                     c("geo_type", "time_type"))) {
               Warn("`additional_metadata` names overlap with existing metadata fields \"geo_type\", \"time_type\".")
             }
-
+            
+            # Finish off with compactify
+            if (missing(compactify)) {
+              compactify = NULL
+            } else if (!rlang::is_bool(compactify) &&
+                        !rlang::is_null(compactify)) {
+              Abort("compactify must be boolean or null.")
+            }
+            
             # Create the data table; if x was an un-keyed data.table itself,
             # then the call to as.data.table() will fail to set keys, so we
             # need to check this, then do it manually if needed
             key_vars = c("geo_value", "time_value", other_keys, "version")
             DT = as.data.table(x, key = key_vars)
             if (!identical(key_vars, key(DT))) setkeyv(DT, cols = key_vars)
-
+            
+            # Checks to see if a value in a vector is LOCF
+            is_locf <- function(vec) {
+              ifelse(!is.na(vec) & !is.na(lag(vec)),
+                     vec == lag(vec),
+                     is.na(vec) & is.na(lag(vec)))
+            }
+            
+            # Checks for LOCF's in a data frame
+            rm_locf <- function(df) {
+             filter(df,if_any(c(everything(),-version),~ !is_locf(.))) 
+            }
+            
+            # Keeps LOCF values, such as to be printed
+            keep_locf <- function(df) {
+              filter(df,if_all(c(everything(),-version),~ is_locf(.))) 
+            }
+            
+            # Runs compactify on data frame
+            if (is.null(compactify) || compactify == TRUE) {
+              elim = keep_locf(DT)
+              DT = rm_locf(DT)
+            } else {
+              # Create empty data frame for nrow(elim) to be 0
+              elim = tibble()
+            }
+            
+            # Warns about redundant rows
+            if (is.null(compactify) && nrow(elim) > 0) {
+              warn <- paste("\nLOCF rows found. To remove warning, set",
+              "compactify to TRUE or fix these rows: \n")
+              # call elim with for loop, up to 6
+              for (i in 1:min(6,nrow(elim))) {
+                warn <- paste0(warn,
+                  paste(elim[[i,1]],elim[[i,2]],elim[[i,3]],elim[[i,4]],"\n")
+                )
+              }
+              if (nrow(elim) > 6) {
+                warn <- paste0(warn,"And so on...")
+              }
+              warning(warn)
```

Comment:
We want to use rlang conditions for errors, warnings, and messages in this package.  Our helper function `Warn` in its current form might mess up the display of `elim` because it does line (re-)wrapping; I'm working on augmenting it to be more convenient in #102.  But for now, maybe just use `rlang::warn` instead of `warning`, and rename the `warn` to something like `warning_message` to prevent confusion.